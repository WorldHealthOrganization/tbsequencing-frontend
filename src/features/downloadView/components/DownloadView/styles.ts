import { css } from '@emotion/react';
import { appZIndex } from '../../../../styles/appZIndex';
import appColors from '../../../../styles/colors';
import { fontSizes } from '../../../../components/typography/fontSizes';

const wrapper = css({
  padding: '40px 54px',
});

const header = css({
  marginBottom: '24px',
});

const sxTabsStyles = {
  '& .MuiTabs-indicator': {
    display: 'flex',
    justifyContent: 'center',
    height: 4,
    zIndex: appZIndex.appBar,
  },
};

const tabsWrapper = css({
  position: 'relative',
  display: 'inline-flex',
  marginBottom: 24,
});

const fakeBorderBottom = css({
  height: 4,
  backgroundColor: appColors.tints.whoBlue.lightest,
  position: 'absolute',
  bottom: 0,
  left: 0,
  right: 0,
});

export const paragraph = css({
  marginBottom: 30,
  color: appColors.neutral.black,
  boxShadow: `0px 0px 4px ${appColors.tints.whoBlue.lighter}`,
  padding: 24,
  fontSize: 16,
  textAlign: 'justify',
});

export const sxTabStyles = {
  fontSize: fontSizes.h2,
  maxWidth: 600,
  padding: `${16}px`,
  color: appColors.neutral.black,
  opacity: 0.8,
  '&.Mui-selected': {
    color: appColors.neutral.black,
    fontWeight: 'bold',
    opacity: 1,
  },
  textTransform: 'none',
};

export {
  wrapper, header, sxTabsStyles, tabsWrapper, fakeBorderBottom,
};
