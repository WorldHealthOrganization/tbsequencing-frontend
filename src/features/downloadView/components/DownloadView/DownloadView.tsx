/** @jsxImportSource @emotion/react */
import React, {
  useState, useEffect, useCallback, SyntheticEvent,
} from 'react';
import { Tabs, Tab } from '@mui/material';
import H1 from '../../../../components/typography/H1';
// import DescriptionText from '../DescriptionText/DescriptionText';
import DrugsDataset from '../DrugsDataset/DrugsDataset';
import MutationsCatalogue from '../MutationsCatalogue/MutationsCatalogue';
import { useGetDrugsDataQuery } from '../../../../services/drugsApi/drugsApi';
import { header, wrapper, sxTabsStyles } from './styles';
import * as styles from './styles';
import { IExtendedDrug } from '../../models';

interface DownloadViewProps {
  mockDate?: string;
}

const DownloadView = ({ mockDate }: DownloadViewProps) => {
  const [tabValue, setTabValue] = useState<number>(1);
  const [drugsList, setDrugsList] = useState<IExtendedDrug[] | []>([]);

  const handleTabChange = useCallback((event: SyntheticEvent, newValue: number) => {
    setTabValue(newValue);
  }, []);

  const handleChangeCheckbox = (touchedItem: IExtendedDrug, checked: boolean) => {
    const touchedDrug = {
      ...touchedItem, isChecked: !checked,
    };
    const nextDrugsList = drugsList.map((item) => ((touchedItem.drugId === item.drugId)
      ? touchedDrug : item));
    setDrugsList(nextDrugsList);
  };

  const handleCheckAll = (isSelect: boolean) => {
    const nextDrugsList = drugsList.map((item) => ({ ...item, isChecked: isSelect }));
    setDrugsList(nextDrugsList);
  };

  const {
    data: drugsData = [],
  } = useGetDrugsDataQuery({});

  useEffect(() => {
    if (drugsData.length) {
      setDrugsList(drugsData.map((item) => ({
        ...item,
        isChecked: false,
      })));
    }
  }, [drugsData]);

  return (
    <div css={wrapper}>
      <H1 style={header}>Data Download</H1>
      {/* <DescriptionText /> */}
      <div css={styles.tabsWrapper}>
        <Tabs value={tabValue} onChange={handleTabChange} sx={sxTabsStyles}>
          <Tab label="WHO Mutations Catalogue 2023 version" value={1} sx={styles.sxTabStyles} />
          <Tab label="Drug resistance associated allelic counts" value={2} sx={styles.sxTabStyles} />
        </Tabs>
        <div css={styles.fakeBorderBottom} />
      </div>
      {tabValue === 2 && (
      <DrugsDataset
        drugsList={drugsList}
        handleChange={handleChangeCheckbox}
        handleCheckAll={handleCheckAll}
        mockDate={mockDate}
      />
      )}
      {tabValue === 1 && <MutationsCatalogue />}
    </div>
  );
};

export default DownloadView;
