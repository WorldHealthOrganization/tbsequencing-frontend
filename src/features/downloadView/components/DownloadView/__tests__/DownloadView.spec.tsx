import React from 'react';
import {
  getTreeJSONWithRedux, render, screen, wrappedInRouter,
} from '../../../../../utils/testUtils/jestUtils';
import DownloadView from '../DownloadView';

describe('DownloadView', () => {
  const props = {
    mockDate: '2023-04-21',
  };

  test('renders DownloadView component without error', () => {
    render(
      wrappedInRouter(<DownloadView {...props} />),
    );
  });

  test('renders DownloadView title without error', () => {
    render(
      wrappedInRouter(<DownloadView {...props} />),
    );

    screen.getByText('Data Download');
  });

  test('Should render correctly', () => {
    const tree = getTreeJSONWithRedux(
      wrappedInRouter(<DownloadView {...props} />),
    );

    expect(tree).toMatchSnapshot();
  });
});
