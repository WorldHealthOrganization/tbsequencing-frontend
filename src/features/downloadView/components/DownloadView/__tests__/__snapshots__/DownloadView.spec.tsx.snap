// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DownloadView Should render correctly 1`] = `
<div
  className="css-1cilyqj-DownloadView"
>
  <p
    className="MuiTypography-root MuiTypography-body1 css-euu0lb-MuiTypography-root-BaseText"
  >
    Data Download
  </p>
  <div
    className="css-1lw751n-DownloadView"
  >
    <div
      className="MuiTabs-root css-1652jll-MuiTabs-root"
    >
      <div
        className="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
        onScroll={[Function]}
        style={
          Object {
            "marginBottom": -0,
            "overflow": "hidden",
          }
        }
      >
        <div
          aria-orientation={null}
          className="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
          onKeyDown={[Function]}
          role="tablist"
        >
          <button
            aria-selected={true}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1r4f6ot-MuiButtonBase-root-MuiTab-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={0}
            type="button"
          >
            WHO Mutations Catalogue version 2
            <span
              className="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
              style={Object {}}
            />
          </button>
          <button
            aria-selected={false}
            className="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1r4f6ot-MuiButtonBase-root-MuiTab-root"
            disabled={false}
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            role="tab"
            tabIndex={-1}
            type="button"
          >
            Drug resistance associated allelic counts
          </button>
        </div>
      </div>
    </div>
    <div
      className="css-6bdnpd-DownloadView"
    />
  </div>
  <div
    className="css-x3ah86-MutationsCatalogue"
  >
    <div
      className="css-12mco17-MutationsCatalogue"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 css-15lj30p-MuiTypography-root-BaseText"
      >
        Download files
      </p>
      <div
        className="css-plr8i0-MutationsCatalogue"
      >
        <p
          className="MuiTypography-root MuiTypography-body1 css-vtlwgn-MuiTypography-root-BaseText"
        >
          Please find below all outputs of the Mutation catalogue 2023 update. These files are identical to those available from the WHO website.
        </p>
        <div>
          <a
            download="Report"
            href="data:text/csv;charset=utf-8,﻿email-sent.png"
            onClick={[Function]}
            target="_self"
          />
          <a
            download="Excel tables"
            href="data:text/csv;charset=utf-8,﻿email-sent.png"
            onClick={[Function]}
            target="_self"
          />
          <a
            download="VCF file"
            href="data:text/csv;charset=utf-8,﻿email-sent.png"
            onClick={[Function]}
            target="_self"
          />
        </div>
        <button
          className="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-disableElevation MuiButtonBase-root css-1ve9vu2-MuiButtonBase-root-MuiButton-root-AppButton"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <p
            className="MuiTypography-root MuiTypography-body1 css-3cwtjx-MuiTypography-root-BaseText"
          >
            Download All
          </p>
        </button>
      </div>
    </div>
    <div
      className="css-1mhavnj-MutationsCatalogue"
    >
      <div
        className="css-15a8d1j-MutationsCatalogue"
      >
        <div
          className="css-yicdw8-MutationsCatalogue"
        >
          Report
        </div>
        <div
          className="css-ftu0pv-MutationsCatalogue"
        >
          <div
            className="css-1711uy7-MutationsCatalogue"
          >
            PDF report of the Mutation Catalogue 2023 update initiative
          </div>
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
            download="w3logo"
            href="email-sent.png"
            onBlur={[Function]}
            onFocus={[Function]}
            target="_blank"
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18vxxzc-MuiSvgIcon-root-MutationsCatalogue"
                data-testid="FileDownloadIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
                />
              </svg>
            </button>
          </a>
        </div>
      </div>
      <div
        className="css-15a8d1j-MutationsCatalogue"
      >
        <div
          className="css-yicdw8-MutationsCatalogue"
        >
          Excel tables
        </div>
        <div
          className="css-ftu0pv-MutationsCatalogue"
        >
          <div
            className="css-1711uy7-MutationsCatalogue"
          >
            Data tables of raw output from the Mutation Catalogue 2023 update initiative
          </div>
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
            download="w3logo"
            href="email-sent.png"
            onBlur={[Function]}
            onFocus={[Function]}
            target="_blank"
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18vxxzc-MuiSvgIcon-root-MutationsCatalogue"
                data-testid="FileDownloadIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
                />
              </svg>
            </button>
          </a>
        </div>
      </div>
      <div
        className="css-15a8d1j-MutationsCatalogue"
      >
        <div
          className="css-yicdw8-MutationsCatalogue"
        >
          VCF file
        </div>
        <div
          className="css-ftu0pv-MutationsCatalogue"
        >
          <div
            className="css-1711uy7-MutationsCatalogue"
          >
            Variant Call Format (VCF) file linking genomic coordinates to variant categories.
          </div>
          <a
            className="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1ps4owl-MuiTypography-root-MuiLink-root"
            download="w3logo"
            href="email-sent.png"
            onBlur={[Function]}
            onFocus={[Function]}
            target="_blank"
          >
            <button
              className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              disabled={false}
              onBlur={[Function]}
              onContextMenu={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={0}
              type="button"
            >
              <svg
                aria-hidden={true}
                className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-18vxxzc-MuiSvgIcon-root-MutationsCatalogue"
                data-testid="FileDownloadIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M19 9h-4V3H9v6H5l7 7 7-7zM5 18v2h14v-2H5z"
                />
              </svg>
            </button>
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
